### ISSUE TITLE
Entire curriculum: Broken link tester

### ISSUE THREAD

**Issue by damon314159** [TheOdinProject/curriculum#27191]
### Checks

- [X] This is not a duplicate of an existing issue (please have a look through our [open issues list](https://github.com/TheOdinProject/curriculum/issues) to make sure)
- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [ ] Would you like to work on this issue?

### Describe your suggestion

A discussion was had yesterday in discord about the feasibility of testing for broken links. There exist some tools that do this job (see example [1](https://www.npmjs.com/package/broken-link-checker) and [2](http://www.linklint.org)). 

These could potentially be integrated into a github workflow much like the recent md linter, in order to replace links faster than waiting for user reports. I think this is most applicable to lesson content, but could also apply to project submissions if a need arises. People are liable to delete old repos or make private, things like that.

Thoughts welcome, but there were [implementation concerns](https://discord.com/channels/505093832157691914/505093832157691916/1199163010032349185) raised by Mclizee when we were talking about this that definitely hold weight

### Path

Other / NA

### Lesson Url

Entire curriculum

**Comment by JoshDevHub** [comment]
This is definitely an interesting idea.

I agree that the best application is probably with quickly catching dead links in the lesson content. I'll bring this up to the team.

**Comment by mao-sz** [comment]
Project submissions probably not practical. We *could* add a link tester action (a search brings up a fair few with many users), but at the same time, given links across the curriculum go temporarily down somewhat often, will such an action be *that* helpful? Vs just letting them be reported organically?

Currently we have a "broken link" issue template that specifies to give it a few hours/a day or so before reporting a link as broken, to ensure it's not temporarily down. And there are loads of webarchive links across the curriculum too.

**Comment by JoshDevHub** [comment]
Yeah I think having an action that runs on every PR probably wouldn't be the best way to handle this. Sites go down frequently, and it isn't always indicative of permanent issues or whatever. Even GitHub goes down from time to time.

I think one thing that could be done (maybe on the main site? maybe with the discord bot?) is a job that runs once a day or so and scans the curriculum for dead links. Any link that's been dead for maybe a week or so is flagged and brought to our attention. I think that kind of thing could be pretty helpful. Especially for links that occur later in the curriculum where catching them via community reporting is going to be less reliable due to fewer people working through that content.

**Comment by mao-sz** [comment]
Hmmmm. Feature in the bot would allow it to easily post in a maint-only channel in any such situations, though IIRC the bot repo doesn't actually have DB stuff itself and just interacts with the website DB for points? Since that's where it'd also have to track dead links. TBH I've always found that system a little odd and has made it tricky to do points-based work locally due to no dev token for points, vs having the bot repo manage its own DB. But that'd be a rather big thing to implement ofc 😩

**Comment by wise-king-sullyman** [comment]
Thanks for taking the time to create this issue, but the maintainer team has discussed the proposal and at this time we don't think this is an automation that we really need.
